src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py

```diff
@@ -272,6 +279,41 @@ def getDatasetLocations(self, datasets, account=None):
                                                              account=account)
             result[datasetPath] = self.cric.PNNstoPSNs(locations)
         return result
+    
+    def getDatasetLocationsFromMSPileup(self, datasetsWithDbsURL):
+        """
+        Returns a dictionary with the location of the datasets according to MSPileup
+        The definition of "location" here is a union of all sites holding at least
+        part of the dataset (defined by the DATASET grouping).
+        :param datasets: list of datasets
+        :param account: rucio account to use, if it is not provided we fallback to
+        pileup account for backward compatibility
+        :return: a dictionary of dataset locations, key'ed by the dataset name
+        """
+        
+        result = {}
+        for dbsUrl, dataset in datasetsWithDbsURL:
+            try:
+                queryDict = {'query': {'pileupName': dataset},
+                             'filters': ['expectedRSEs', 'currentRSEs', 'pileupName', 'containerFraction', 'ruleIds']}
+                pileUpinstance = '-testbed' if 'cmsweb-testbed' in dbsUrl else '-prod'
+                msPileupUrl = f"https://cmsweb{pileUpinstance}.cern.ch/ms-pileup/data/pileup"
+                self.logger.info(f'dbsUrl: {dbsUrl}')
+                self.logger.info(f'msPileupUrl: {msPileupUrl}')
+                doc = getPileupDocs(msPileupUrl, queryDict, method='POST')[0]
+                currentRSEs = doc['currentRSEs']
+                self.logger.info(f'doc: {doc}')
+                if len(currentRSEs) == 0:
+                    self.logger.warning(f'No RSE has a copy of the desired pileup dataset. Expected RSEs: {doc["expectedRSEs"]}')  
+                self.logger.info(f'locationsFromPileup - name: {dataset}, currentRSEs: {doc["currentRSEs"]}, containerFraction: {doc["containerFraction"]}')
+                result[dataset] = doc['currentRSEs']
+            except IndexError:
+                self.logger.warning('Did not find any pileup document for query: %s', queryDict['query'])
+                continue
```

Comment:
I am really glad I had a careful look at these unit tests, otherwise a nasty bug would sneak through.
Please add
```
                result[dataset] = []  # equal to an empty currentRSEs
```
before the `continue` statement. Otherwise, workqueue elements would be completely missing the pileup information. This fixes one of the unit tests.